deepchem/models/torch_models/layers.py

```diff
@@ -6164,3 +6164,182 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
         output = H_out * T_out + x * (1 - T_out)
 
         return output
+
+
+def cosine_dist(x, y, eps=1e-8):
```

Comment:
1. Refer to Pytorch docs for the cosine similarity and correct the epsilon term in code.
2. Check is this layer uses cosine distance or cosine similarity, refer to the research paper

Reply:
went through the paper it's cosine distance